apps/graduate/src/features/all-management/components/UserDetailModal/UserDetailModal.tsx

```diff
@@ -0,0 +1,58 @@
+import { Descriptions, Modal } from 'antd';
+import { useState } from 'react';
+
+import { Header } from '~/shared/components';
+import { Container } from '~/shared/components/Container';
+
+import UserApplicationView from './UserApplicationView';
+import UserDetailView from './UserDetailView';
+import UserFinalReportView from './UserFinalReportView';
+import UserMiddleReportView from './UserMiddleReportView';
+import { userDetailData } from '../../mock/allManagement';
+import type { Mode } from '../../types/allManagement';
+import { MODE_SUBTITLES } from '../../types/allManagement';
+
+interface UserDetailModalProps {
+  isModalOpen: boolean;
+  setIsModalOpen: (isModalOpen: boolean) => void;
+  // selectedId: string;
+}
+
+export default function UserDetailModal({
+  isModalOpen,
+  setIsModalOpen,
+}: UserDetailModalProps) {
+  const [mode, setMode] = useState<Mode>('detail');
+  const { studentId, period, name, professor, department, delay, etc } =
+    userDetailData;
+
+  return (
+    <Modal
+      open={isModalOpen}
+      width={600}
+      getContainer={false}
+      onCancel={() => setIsModalOpen(false)}
+      footer={null}
+      centered
+    >
+      <Header title={name} subtitle={MODE_SUBTITLES[mode]} />
+
+      <Container style={{ padding: '0px' }}>
+        <Descriptions column={2} bordered>
+          <Descriptions.Item label='학번'>{studentId}</Descriptions.Item>
+          <Descriptions.Item label='졸업시기'>{period}</Descriptions.Item>
+          <Descriptions.Item label='이름'>{name}</Descriptions.Item>
+          <Descriptions.Item label='지도교수'>{professor}</Descriptions.Item>
+          <Descriptions.Item label='소속학과'>{department}</Descriptions.Item>
+          <Descriptions.Item label='지연횟수'>{delay}</Descriptions.Item>
+          <Descriptions.Item label='기타자격'>{etc}</Descriptions.Item>
+        </Descriptions>
+      </Container>
+
+      {mode === 'detail' && <UserDetailView setMode={setMode} />}
+      {mode === 'application' && <UserApplicationView setMode={setMode} />}
+      {mode === 'middleReport' && <UserMiddleReportView setMode={setMode} />}
+      {mode === 'finalReport' && <UserFinalReportView setMode={setMode} />}
```

Comment:
이 부분 요렇게 하기보다는 상단에 모드에 따른 `View`를 반환해주는 함수를 만들어서 사용하는 것이 어떤 모드에 어떤 뷰가 해당하는지 더 잘 알 수 있을 것 같아요!

```suggestion
      const renderViewByMode = (mode: Mode) => {
          if (mode === 'detail') return <UserDetailView setMode={setMode} />
          ...
          }
      // case 문으로 작성해도 좋을 것 같아요
      ...
      {renderViewByMode(mode)}
```

Reply:
또는 제가 위에서 언급한 [context API](https://react.dev/learn/passing-data-deeply-with-context)를 사용해서 상태, 상태에 따른 `View` 를 모두 분리해서 관리하는 것도 좋을 것 같아요. `UserDetailModal` 컴포넌트가 유저 상세 모달 자체에 대한 단일 책임을 가지게끔 리팩토링 할 수 있어보여 코멘트 달아요!